enhancements/two-nodes-openshift/2no.md

```diff
@@ -113,32 +119,57 @@ In both cases, the control-plane's dependence on etcd will cause it to respond w
+The requirement that the cluster can be deployed using only 2 nodes is key because requiring a third baremetal server for installation can be expensive when deploying baremetal at scale. To accomplish this, deployments will use one of the target machines as the bootstrap node before it is rebooted into a control-plane node.
+
+A critical transition during bootstrapping is when the bootstrap reboots into the control plane node. Before this reboot, it needs to be removed from the etcd cluster so that quorum can be maintained as the machine reboots into a second control-plane.
+
+Otherwise, the procedure follows the standard flow except for the configuration of 2 nodes instead of 3.
+
+Because BMC passwords are being collected to initialize fencing, the SaaS offering will not be available (to avoid storing customer BMC credentials in a Red Hat database). ZTP may be considered in the future, but this will need further evaluation to ensure passwords are appropriately handled.
+
+Everything else about cluster creation will be an opaque implementation detail not exposed to the user.
+
+##### Transitioning to a RHEL-HA Controlled Cluster
+Three aspects of cluster creation need to happen for a vanilla two-node cluster to have RHEL-HA functioning as described in the proposal.
+1. Initializing the RHEL-HA cluster
+2. Transitioning control of etcd to RHEL-HA
+3. Enabling fencing in RHEL-HA
 
-Otherwise, the procedure follows the standard flow except for the configuration of 2 nodes instead of 3. At this time we've discussed the collection of RedFish details (including passwords!) for each node. This is needed for the RHEL-HA configuration by leveraging the BareMetalHost CRDs populated from the baremetal platform specification in the install-config. There are open questions on how to ensure that pacemaker is the only entity responsible for fencing to prevent conflicting requests to change the machine state between pacemaker and the baremetal operator. Preventing conflicting fencing logic is also important for optional operators like Node Health Check, Self Node Remediation, and Fence Agents Remediation, but these should not be present during installation.
+We propose the inclusion of a new in-cluster operator responsible for the remediation of 2NO resource agents.
 
-An important facility of the installation flow is the transition from a CEO deployed etcd to one controlled by pacemaker. The basic transition works as follows:
+###### Transitioning Etcd Management to RHEL-HA
+An important facility of the installation flow is the transition from a CEO deployed etcd to one controlled by RHEL-HA. The basic transition works as follows:
 1. MCO Extensions are used to ensure that pacemaker, corosync, and resource agents are pre-configured on CoreOS using installation manifests.
```

Comment:
Can you elaborate on `MCO Extensions` and `installation manifests`?
Does it mean that MCO will generate these manifests (similar to [this](https://github.com/openshift/machine-config-operator/blob/67789532e4569ece95eb98fed1dde03409f065e9/pkg/operator/bootstrap.go#L185)) or we expect the installer / user to provide them as [extraManifests](https://github.com/openshift/installer/blob/2ea001d6d7cb97b1eb081c80b48b1151f2f30257/pkg/asset/agent/manifests/extramanifests.go#L16) for MCO to apply them

Reply:
I will provide some supportive links, but what I mean by mco extensions is this:
https://docs.openshift.com/container-platform/4.17/machine_configuration/machine-configs-configure.html#rhcos-add-extensions_machine-configs-configure
https://github.com/openshift/installer/blob/2ea001d6d7cb97b1eb081c80b48b1151f2f30257/docs/user/customization.md?plain=1#L561

In short - we need to install additional RPMs on top of RHCOS. MCO extensions are a feature that allows the user to install a curated set of packages on top of RHCOS. For the pacemaker & corosync rpms, I think the answer is very much that MCO does that. However, we also need to install some configuration logic for our 2node specific pacemaker. That isn't packaged as an RPM, so I guess it could be either.

@clobrano - any input?

Reply:
> we need to install additional RPMs on top of RHCOS...

that's correct.

> I think the answer is very much that MCO does that

I don't know the details, but if MCO can understand its 2NO by the `platform`, then I tend to agree, otherwise would be the installer responsibility (not the user though)

> However, we also need to install some configuration logic for our 2node specific pacemaker. That isn't packaged as an RPM, so I guess it could be either.

This are MachineConfigs we can install via the operator. Whether it's a custom operator for 2NO (like the one we have for the current demo) or the extension of an already existing operator.

Reply:
I think I captured this. We may want to review the wording in the newest patch.